### PR TITLE
1.13+ update

### PR DESCRIPTION
**Pull request by barpec12**
This PR gives you possibility to run 1.13 and further versions on SKCraft. Enjoy! :)

### CODE REVIEW

**Comment by HonestTrash** [comment]
This really needs to be committed soon!!

**Comment by Routhinator** [comment]
Does this fix the builder? And does this move all versions of MC downloading to the new location? I've noticed that these pulls all target 1.13+ only, but MS is moving ALL downloads to the new location, not just 1.13+. And the old location is due to be removed/deprecated by the end of this month.

**Comment by barpec12** [comment]
@Routhinator These pulls support all versions. Builder works as well.

**Comment by jkroepke** [comment]
@barpec12 But i'm unable to run an 1.13.2 instance on MacOS

```
[information] Building asset virtual tree at '/Users/jan/IdeaProjects/atu-launcher/./assets/virtual/1.13.1'...
[information] Process ended; cleaning up /Users/jan/IdeaProjects/atu-launcher/./temp/extract/1545613825225
[warnung] Task failed
java.lang.NullPointerException
	at com.skcraft.launcher.launch.Runner.addJarArgs(Runner.java:280)
	at com.skcraft.launcher.launch.Runner.call(Runner.java:137)
	at com.skcraft.launcher.launch.Runner.call(Runner.java:45)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Any idea?

File: launcher/src/main/java/com/skcraft/launcher/update/Updater.java [hunk 1]
```diff
@@ -171,7 +168,7 @@ protected void update(Instance instance) throws Exception {
         // Download assets
         log.info("Enumerating assets to download...");
         progress = new DefaultProgress(-1, SharedLocale.tr("instanceUpdater.collectingAssets"));
-        installAssets(installer, version, launcher.propUrl("assetsIndexUrl", version.getAssetsIndex()), assetsSources);
+        installAssets(installer, version, url(version.getAssetIndex().get("url")), assetsSources);
```
Comment (jkroepke) [inline launcher/src/main/java/com/skcraft/launcher/update/Updater.java]:
```suggestion
        URL assetUrl = version.getAssetIndex() != null
                ? url(version.getAssetIndex().get("url"))
                : launcher.propUrl("assetsIndexUrl", version.getAssetsIndex());
        installAssets(installer, version, assetUrl, assetsSources);
```

to support modpack created with old builder? Also assetsIndexUrl in the properties should not delete.

**Comment by barpec12** [comment]
Good idea, I will correct that in a few days.

**Comment by jkroepke** [comment]
Found an other issue. If I compare the versionManifest between https://launchermeta.mojang.com/v1/packages/df66b5a5f451d026c356ead8caf118c71919850b/1.13.2.json and my own I see that `versionManifest.arguments.jvm` is completely missing. Its mandatory for OSX again since MC need the JVM flag `-XstartOnFirstThread` which is only available at OSX.

File: launcher/src/main/java/com/skcraft/launcher/model/minecraft/Arguments.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+/**
+ * @author barpec12
+ */
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Arguments {
+
+    private List<Object> game;
```
Comment (jkroepke) [inline launcher/src/main/java/com/skcraft/launcher/model/minecraft/Arguments.java]: https://github.com/SKCraft/Launcher/pull/270/#issuecomment-449992857
```suggestion
     private List<Object> game;
     private List<Object> jvm;
```

**Comment by jkroepke** [comment]
@barpec12 if its okay I would extend this PR with the features around the arguments?

File: launcher/src/main/java/com/skcraft/launcher/update/Updater.java (hunk 1 above)
Comment (jkroepke) [reply launcher/src/main/java/com/skcraft/launcher/update/Updater.java]: Soon™ 

Mojang deprecate https://s3.amazonaws.com/Minecraft.Download/indexes/%s.json but I would prefer to remove this functionally from this launcher here if the link is really dead.

In mean time older mod pack are still working and need only rebuild if the S3 Endpoint is not available. This is a nice backward compatibility which is important, e.g. for @sk89q (see https://github.com/SKCraft/Launcher/pull/164#issuecomment-254714556)

**Comment by jkroepke** [comment]
> That would be nice.

After some research I think I need a custom deserializer to map the arguments into Java objects becuase the type of the arguments game and arguments jvm isn't constant. 

I can not provide it because my java skill are just limited to stack overflow ...

But i think its important for future compability of minecraft. (For now we need `-XstartOnFirstThread` on OSX otherwise Minecraft will not start anymore)

**Comment by LoneDev6** [comment]
When will this committed? I need to install Fabric 1.14.4 as loader, thanks

**Comment by spannerman79** [comment]
@LoneDev6 you'll need to add this to your own fork until @sk89q updates

**Comment by LoneDev6** [comment]
Added it thanks
I hope anyone will solve the fabric issue

**Comment by SDUBZ** [comment]
How do i PR this into my own build since i didnt directly fork but made from source then made on gh later on after i edited things 🤔

**Comment by SDUBZ** [comment]
@barpec12 whats with all the <<<< head in your files
tried changes you made from officl branch now launcher doesnt even build

**Comment by barpec12** [comment]
@SDUBZ I don't see anything like this. Can you send me a github link to this line?

**Comment by SDUBZ** [comment]
@barpec12 this is what i ran to get the changes 
git remote add 113update https://github.com/barpec12/Launcher
git fetch 113update
git pull 113update official

**Comment by barpec12** [comment]
That's not a bug. You are trying to merge my changes into your repository. Unfortunately there are some conflicts you have to solve manually. Read https://stackoverflow.com/questions/10657315/git-merge-left-head-marks-in-my-files

**Comment by spannerman79** [comment]
@barpec12 PR's are to review and comment on code that may or may not have issues when it comes to the base repo, to raise points specific to the PR to which the PR is being requested to be included within.

A PR is **not a tech support place**. https://help.github.com/en/articles/about-pull-requests

**Comment by hedgehog1029** [comment]
Superseded by the changes in master; thanks anyway!

**Pull request closed without merging**